### PR TITLE
Add new API: getTabsInChannels().

### PR DESCRIPTION
**Pull request by jayongg**
1. Upgrade Typescript to 2.5.0
2. Add new API: getTabsInChannels().

### CODE REVIEW

File: src/MicrosoftTeams.ts [hunk 1]
```diff
@@ -53,7 +53,24 @@ namespace microsoftTeams
         args?: any[]; // tslint:disable-line:no-any:The args here are a passthrough from OnMessage where we do receive any[]
     }
 
-    // This indicates whether initialize was called (started).
+    export interface AppInformation {
+        tabList?: TabInformation[];
+    }
+
+    export interface TabInformation {
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: Rename to TabInstance as discussed

File: src/MicrosoftTeams.ts [hunk 2]
```diff
@@ -53,7 +53,24 @@ namespace microsoftTeams
         args?: any[]; // tslint:disable-line:no-any:The args here are a passthrough from OnMessage where we do receive any[]
     }
 
-    // This indicates whether initialize was called (started).
+    export interface AppInformation {
+        tabList?: TabInformation[];
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: Consider renaming tabList to teamTabs or teamTabsList.

File: src/MicrosoftTeams.ts [hunk 3]
```diff
@@ -53,7 +53,24 @@ namespace microsoftTeams
         args?: any[]; // tslint:disable-line:no-any:The args here are a passthrough from OnMessage where we do receive any[]
     }
 
-    // This indicates whether initialize was called (started).
+    export interface AppInformation {
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: Rename to TabInformation as discussed

File: src/MicrosoftTeams.ts [hunk 4]
```diff
@@ -53,7 +53,24 @@ namespace microsoftTeams
...
+        tabName: string;
+        entityId?: string;
+        channelId?: string;
+        channelName?: string;
+        teamId?: string;
+        teamName?: string;
+        groupId?: string;
+        url?: string;
+        websiteUrl?: string;
+        customSettings?: string;
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: Delete

File: src/MicrosoftTeams.ts [hunk 5]
```diff
@@ -53,7 +53,24 @@ namespace microsoftTeams
...
+    }
+
+        // This indicates whether initialize was called (started).
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: Fix indentation

File: src/MicrosoftTeams.ts [hunk 6]
```diff
@@ -226,6 +243,17 @@ namespace microsoftTeams
     }
 
     /**
+     * Allows an app to retrieve all the tabs in favorite channels where it is enabled for this user
+     */
+    export function getTabsInChannels(callback: (appInfo: AppInformation) => void): void
```
Comment (ydogandjiev) [inline src/MicrosoftTeams.ts]: Rename to getTabInstances as discussed

File: src/MicrosoftTeams.ts (hunk 4 above)
Comment (jayongg) [reply src/MicrosoftTeams.ts]: customSettings removed.

**Pull request merged**